#167706301 Edit User Profile

JuwonAbiola opened the pull request:
#### What does this PR do?
- It implements the feature to edit user profile
- It implements the feature to view user profile
- It implements the feature to view other user profile

#### Description of Task proposed in this pull request?
- create helper to upload images
- update  user controller  to enable  profile editing
- write test for code changes

#### How should this be manually tested (Quality Assurance)?
- Update an existing user by making a PATCH request to api/v1/users/editprofile

#### What are the relevant pivotal tracker stories?
- [167706301](https://www.pivotaltracker.com/story/show/167706301)

#### Any background context you want to add (Operations Impact)?
add the following to the .env file
- CLOUD_NAME
- API_KEY
- API_SECRET


#### What I have learned working on this feature: I have learnt how to upload images using cloudinary 
#### Screenshots: 
![image](https://user-images.githubusercontent.com/42170732/62695605-93779500-b9ce-11e9-900a-d90d9ee02703.png)

henryade commented on server/controllers/Profiles.js:
```diff
@@ -0,0 +1,66 @@
+   * @param {object} req request object
+   * @param {object} res response object
+   * @returns {json} the json response been return by the server
+   * @memberof UserController
```
huh?????

henryade commented on server/controllers/Profiles.js:
```diff
@@ -0,0 +1,66 @@
+const { User } = models;
+
+/**
+ * @export
```
i dont understand

henryade commented on test/profiles/update.test.js:
```diff
@@ -0,0 +1,113 @@
+      email: 'Jhayx@rambo.com',
+      password: 'incorrect',
+      confirmPassword: 'incorrect'
+    };
```
this should come as a mock data

chialuka reviewed (dismissed):
LGTM

dinobi reviewed (changes requested):
Please update the response sent to client to match the one proposed by @abdulfatai360

chialuka commented on server/routes/profile.js:
```diff
@@ -0,0 +1,23 @@
+
+const route = express.Router();
+
+route.patch(
```
The validation middleware is not included here. Please add it to prevent getting a server error when users try to enter invalid values in updating their profiles.

chialuka commented on server/controllers/Profiles.js:
```diff
@@ -0,0 +1,73 @@
+      }
+      await User.update(
+        {
+          ...body,
```
If users can change their emails, then provision should be made to send them verification emails to the new email address they choose. Then the new email should be marked unverified in the DB

JuwonAbiola replied:
Thanks for the feedback I thought of that as well but I think that is beyond the scope of the task assigned.

Adekoreday reviewed (commented):
kindly make this multer middleware more reusable so we can pass in different fields into it.
e.g multer('avatarUrl');

JuwonAbiola committed:
feature-[167706301]:edit user profile
- install cloudinary
- install multer
- write unit test
- create helper to upload image
- write controller method to edit user
- write controller methoid to view user profile
[Delivers #167706301]
